src/main/java/application/onlinebookstore/controller/BookController.java

```diff
@@ -0,0 +1,61 @@
+package application.onlinebookstore.controller;
+
+import application.onlinebookstore.dto.BookDto;
+import application.onlinebookstore.dto.BookSearchParametersDto;
+import application.onlinebookstore.dto.CreateBookRequestDto;
+import application.onlinebookstore.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/api/books")
+public class BookController {
+    private final BookService bookService;

```

Comment:
Book Endpoints:

GET: /api/books (Retrieve book catalog) (should be done in the previous PR)
GET: /api/books/{id} (Retrieve book details) (should be done in the previous PR)
POST: /api/books (Create a new book) (should be done in the previous PR)
PUT: /api/books/{id} (Update a specific book) <----------------------------------------- U miss this